Test/QuickCheck/Monadic.hs

```diff
@@ -105,14 +108,23 @@ newtype PropertyM m a =
 instance Functor (PropertyM m) where
   fmap f (MkPropertyM m) = MkPropertyM (\k -> m (k . f))
 
-instance Monad m => Applicative (PropertyM m) where
-  pure = return
-  (<*>) = liftM2 ($)
+instance Applicative (PropertyM m) where
+  pure x  = MkPropertyM ($ x)
+  f <*> v = MkPropertyM $ \ c -> unPropertyM f $ \ g -> unPropertyM v (c . g)
+
+#if MIN_VERSION_base(4,9,0)
+instance Monad (PropertyM m) where
+  return x = MkPropertyM ($ x)
```

Comment:
just `return = pure` would be clearer? Or actually omitting it fully in base-4.9 case?